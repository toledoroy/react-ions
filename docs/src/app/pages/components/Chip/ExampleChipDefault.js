import React, { Fragment } from 'react'
import Chip from 'react-ions/lib/components/Chip'

const inlineStyle = {
  marginRight: '2px',
  textTransform: 'capitalize'
}

const chipList = [
  'neutral-1',
  'neutral-2',
  'neutral-3',
  'neutral-4',
  'primary-1',
  'primary-2',
  'primary-3',
  'primary-4',
  'primary-5',
  'primary-6',
  'danger',
  'warning',
  'success',
  'info',
]

const chips = () => {
  return chipList.map(chip => {
    return <Chip text={chip} color={chip} style={inlineStyle} />
  })
}

const ExampleChipDefault = () => (
  <Fragment>
    {chips()}
  </Fragment>
)

export default ExampleChipDefault
