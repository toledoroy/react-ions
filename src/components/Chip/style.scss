@import '../../styles/common/colors';

.chip-wrapper {
  background-color: $neutral-4;
  border-radius: 2px;
  color: $primary-4;
  display: inline-flex;
  font-size: 12px;
  font-weight: normal;
  padding: 3px 12px;

  span {
    line-height: 1.6;
    position: relative;
  }

  &.larger {
    font-size: 14px;
  }

  &.smaller {
    font-size: 10px;
    font-weight: 200;
    padding: 5px 8px 3px;
    text-transform: uppercase;
  }

  &.primary, &.primary-1 {
    background-color: $primary-1;
    color: $white;
  }

  &.primary-2 {
    background-color: $primary-2;
    color: $white;
  }

  &.primary-3 {
    background-color: $primary-3;
    color: $white;
  }

  &.primary-4 {
    background-color: $primary-4;
    color: $white;
  }

  &.primary-5 {
    background-color: $primary-5;
    color: $white;
  }

  &.primary-darker, &.primary-6 {
    background-color: $primary-6;
    color: $white;
  }

  &.neutral-1 {
    background-color: $neutral-1;
  }

  &.neutral-2 {
    background-color: $neutral-2;
  }

  &.neutral-3 {
    background-color: $neutral-3;
  }

  &.neutral-4 {
    background-color: $neutral-4;
  }

  &.danger {
    background-color: $danger;
    color: $white;
  }

  &.success {
    background-color: $success;
    color: $white;
  }

  &.warning {
    background-color: $warning;
    color: $white;
  }

  &.info {
    background-color: $info;
    color: $white;
  }

  &.is-actionable {
    cursor: pointer;
  }
}
