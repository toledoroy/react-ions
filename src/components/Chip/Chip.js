import React from 'react'
import PropTypes from 'prop-types'
import optclass from '../internal/OptClass'
import style from './style.scss'

const Chip = props => {
  const isActionable = props.clickCallback ? 'is-actionable' : null
  const chipClasses = optclass(style, ['chip-wrapper', props.color, props.size, props.optClass, props.className, isActionable])

  return (
    <span className={chipClasses} style={props.style}>
      <span onClick={props.clickCallback}>{props.text}</span>
    </span>
  )
}

Chip.propTypes = {
  /**
   * The background color of the chip (see Foundations > Colors for more info).
   */
  color: PropTypes.oneOf([
    'primary',
    'primary-1',
    'primary-2',
    'primary-3',
    'primary-4',
    'primary-5',
    'primary-darker',
    'primary-6',
    'neutral-1',
    'neutral-2',
    'neutral-3',
    'neutral-4',
    'danger',
    'success',
    'warning',
    'info'
  ]),
  /**
   * Optional click callback.
   */
  clickCallback: PropTypes.func,
  /**
   * Optional styles to add to the chip.
   */
  optClass: PropTypes.string,
  /**
   * Optional size of the chip.
   */
  size: PropTypes.oneOf(['larger', 'smaller']),
  /**
   * Optional style object to be added for additional CSS.
   */
  style: PropTypes.object,
  /**
   * Text to display inside the chip.
   */
  text: PropTypes.string
}

Chip.defaultProps = {
  color: 'primary'
}

export default Chip
