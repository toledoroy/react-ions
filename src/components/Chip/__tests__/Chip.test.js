import React from 'react'
import Chip from '../Chip'

describe('Chip', () => {
  it('should render with the primary color', () => {
    const chip = shallow(<Chip color='primary' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with the primary-1 color', () => {
    const chip = shallow(<Chip color='primary-1' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with the primary-2 color', () => {
    const chip = shallow(<Chip color='primary-2' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with the primary-3 color', () => {
    const chip = shallow(<Chip color='primary-3' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with the primary-4 color', () => {
    const chip = shallow(<Chip color='primary-4' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with the primary-5 color', () => {
    const chip = shallow(<Chip color='primary-5' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with the primary-darker color', () => {
    const chip = shallow(<Chip color='primary-darker' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with the primary-6 color', () => {
    const chip = shallow(<Chip color='primary-6' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with the neutral-1 color', () => {
    const chip = shallow(<Chip color='neutral-1' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with the neutral-2 color', () => {
    const chip = shallow(<Chip color='neutral-2' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with the neutral-3 color', () => {
    const chip = shallow(<Chip color='neutral-3' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with the neutral-4 color', () => {
    const chip = shallow(<Chip color='neutral-4' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with the danger color', () => {
    const chip = shallow(<Chip color='danger' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with the success color', () => {
    const chip = shallow(<Chip color='success' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with the warning color', () => {
    const chip = shallow(<Chip color='warning' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with the info color', () => {
    const chip = shallow(<Chip color='info' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with a className', () => {
    const chip = shallow(<Chip className='classy-class' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with text', () => {
    const chip = shallow(<Chip text='text' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with a larger size', () => {
    const chip = shallow(<Chip size='larger' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with a smaller size', () => {
    const chip = shallow(<Chip size='smaller' />)
    expect(chip).toMatchSnapshot()
  })

  it('should render with a style object', () => {
    const chip = shallow(<Chip size='smaller' style={{marginRight: '4px'}} />)
    expect(chip).toMatchSnapshot()
  })
})
