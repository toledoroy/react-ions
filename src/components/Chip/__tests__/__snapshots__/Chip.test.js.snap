// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip should render with a larger size 1`] = `
<span
  className="chip-wrapper primary larger"
>
  <span />
</span>
`;

exports[`Chip should render with a smaller size 1`] = `
<span
  className="chip-wrapper primary smaller"
>
  <span />
</span>
`;

exports[`Chip should render with an optClass 1`] = `
<span
  className="chip-wrapper primary classy-class"
>
  <span />
</span>
`;

exports[`Chip should render with text 1`] = `
<span
  className="chip-wrapper primary"
>
  <span>
    text
  </span>
</span>
`;

exports[`Chip should render with the danger color 1`] = `
<span
  className="chip-wrapper danger"
>
  <span />
</span>
`;

exports[`Chip should render with the info color 1`] = `
<span
  className="chip-wrapper info"
>
  <span />
</span>
`;

exports[`Chip should render with the neutral-1 color 1`] = `
<span
  className="chip-wrapper neutral-1"
>
  <span />
</span>
`;

exports[`Chip should render with the neutral-2 color 1`] = `
<span
  className="chip-wrapper neutral-2"
>
  <span />
</span>
`;

exports[`Chip should render with the neutral-3 color 1`] = `
<span
  className="chip-wrapper neutral-3"
>
  <span />
</span>
`;

exports[`Chip should render with the neutral-4 color 1`] = `
<span
  className="chip-wrapper neutral-4"
>
  <span />
</span>
`;

exports[`Chip should render with the primary color 1`] = `
<span
  className="chip-wrapper primary"
>
  <span />
</span>
`;

exports[`Chip should render with the primary-1 color 1`] = `
<span
  className="chip-wrapper primary-1"
>
  <span />
</span>
`;

exports[`Chip should render with the primary-2 color 1`] = `
<span
  className="chip-wrapper primary-2"
>
  <span />
</span>
`;

exports[`Chip should render with the primary-3 color 1`] = `
<span
  className="chip-wrapper primary-3"
>
  <span />
</span>
`;

exports[`Chip should render with the primary-4 color 1`] = `
<span
  className="chip-wrapper primary-4"
>
  <span />
</span>
`;

exports[`Chip should render with the primary-5 color 1`] = `
<span
  className="chip-wrapper primary-5"
>
  <span />
</span>
`;

exports[`Chip should render with the primary-6 color 1`] = `
<span
  className="chip-wrapper primary-6"
>
  <span />
</span>
`;

exports[`Chip should render with the primary-darker color 1`] = `
<span
  className="chip-wrapper primary-darker"
>
  <span />
</span>
`;

exports[`Chip should render with the success color 1`] = `
<span
  className="chip-wrapper success"
>
  <span />
</span>
`;

exports[`Chip should render with the warning color 1`] = `
<span
  className="chip-wrapper warning"
>
  <span />
</span>
`;
